### PR TITLE
add usb_reset

### PR DESCRIPTION
**Pull request by k-okada**
Hi

Sometimes, openni2 driver is not able to open or connect to device and we have to physically re-connect the USB cable to reset usb status, This program (usb_reset) is do similar effect with software.

if you know the usb device path using `lsusb` command, then you can try

```
rosrun openni2_camera usb_reset /dev/bus/usb/002/017
```

Cc: @garaemon

### CODE REVIEW

**Comment by garaemon** [comment]
As far as I know, openni_camera has the same problem.

2014年10月2日木曜日、Kei Okadanotifications@github.comさんは書きました:

> Hi
> 
> Sometimes, we have on openni2 driver, which the program is not able to
> open or connect to device and we have to physically re-connect the USB
> cable to reset usb status, This program (usb_reset) is do similar effect
> with software.
> 
> if you know the usb device path using lsusb command, then you can try
> 
> rosrun openni2_camera usb_reset /dev/bus/usb/002/017
> 
> ## Cc: @garaemon https://github.com/garaemon
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/k-okada/openni2_camera add_usb_reset
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/ros-drivers/openni2_camera/pull/21
> Commit Summary
> - add usb_reset
> 
> File Changes
> - _M_ CMakeLists.txt
>   https://github.com/ros-drivers/openni2_camera/pull/21/files#diff-0
>   (2)
> - _A_ src/usb_reset.c
>   https://github.com/ros-drivers/openni2_camera/pull/21/files#diff-1
>   (71)
> 
> Patch Links:
> - https://github.com/ros-drivers/openni2_camera/pull/21.patch
> - https://github.com/ros-drivers/openni2_camera/pull/21.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-drivers/openni2_camera/pull/21.

## 

from iPhone

File: CMakeLists.txt [hunk 1]
```diff
@@ -57,6 +57,8 @@ add_executable(openni2_camera_node
 target_link_libraries(openni2_camera_node openni2_driver_lib ${catkin_LIBRARIES} ${Boost_LIBRARIES} )
 add_dependencies(openni2_camera_node ${PROJECT_NAME}_gencfg)
 
+add_executable(usb_reset src/usb_reset.c)
+
 install(TARGETS openni2_wrapper openni2_camera_nodelet openni2_camera_node openni2_driver_lib
```
Comment (jonbinney) [inline CMakeLists.txt]: Probably want to install usb_reset as well?

Comment (k-okada) [inline CMakeLists.txt]: yes, I fixed them. see https://github.com/k-okada/openni2_camera/commit/34977c1887fe35c72a395eb7dd8da6dfa32d5526

**Comment by mikeferguson** [comment]
Looks useful, in the future I imagine you might want to put this in a generic "usb_utils" package or something like that, but we can include it in openni2 for now.

**Comment by mikeferguson** [comment]
@k-okada Can you also open a PR to put this in Indigo? Thx

**Comment by k-okada** [comment]
Ok, added a PR for indigo https://github.com/ros-drivers/openni2_camera/pull/22

**Comment by jihoonl** [comment]
It seems nice features for all linux system. How about moving this to https://github.com/ros-drivers/linux_peripheral_interfaces?

**Pull request merged**